File: music-application/src/main/java/com/music/sale/application/product/port/inport/ProductCommandUseCase.java
package com.music.sale.application.product.port.inport;

import com.music.sale.application.product.dto.input.CreateProductInput;
import com.music.sale.application.product.dto.input.UpdateProductInput;
import com.music.sale.application.product.dto.output.CreateProductOutput;
import com.music.sale.application.product.dto.output.UpdateProductOutput;

public interface ProductCommandUseCase {
    CreateProductOutput createProduct(CreateProductInput input, Long currentUserId);
    UpdateProductOutput updateProduct(Long productId, UpdateProductInput input, Long currentUserId);
    void deleteProduct(Long productId, Long currentUserId);
}

File: music-application/src/main/kotlin/com/music/sale/application/product/port/inport/ProductCommandUseCase.kt
package com.music.sale.application.product.port.inport

import com.music.sale.application.product.dto.CreateProductInput
import com.music.sale.application.product.dto.ProductOutput
import com.music.sale.application.product.dto.UpdateProductInput

interface ProductCommandUseCase {
    fun createProduct(input: CreateProductInput): ProductOutput

    fun updateProduct(input: UpdateProductInput): ProductOutput

    fun deleteProduct(id: Long): ProductOutput
} 
